app/src/org/odk/collect/android/tasks/SaveToDiskTask.java

```diff
@@ -89,30 +91,46 @@ public SaveToDiskTask(Uri mUri, Boolean saveAndExit, Boolean markCompleted, Stri
      * an instance, it will be used to fill the {@link FormDef}.
      */
     @Override
-    protected SaveStatus doTaskBackground(Void... nothing) {
+    protected Pair<SaveStatus, String> doTaskBackground(Void... nothing) {
         if (hasInvalidAnswers(mMarkCompleted, DeveloperPreferences.shouldFireTriggersOnSave())) {
-            return SaveStatus.INVALID_ANSWER;
+            return new Pair<>(SaveStatus.INVALID_ANSWER, "");
         }
 
         FormEntryActivity.mFormController.postProcessInstance();
 
-        if (exportData(mMarkCompleted)) {
-            if (exitAfterSave) {
-                return SaveStatus.SAVED_AND_EXIT;
-            } else if (mMarkCompleted) {
-                return SaveStatus.SAVED_COMPLETE;
-            } else {
-                return SaveStatus.SAVED_INCOMPLETE;
-            }
+        try {
+            exportData(mMarkCompleted);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            return new Pair<>(SaveStatus.SAVE_ERROR,
+                    "Something is blocking acesss to the submission file in " + FormEntryActivity.mInstancePath);
```

Comment:
Are these getting bubbbled up to the user? If so we possibly want them to be localizable...